controller/src/types/workload_request.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+use validator::{Validate, ValidationError};
+
+#[derive(Debug, Validate, Deserialize)]
+pub struct WorkloadRequest {
+    pub version: String,
+    pub workload: Workload,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum WorkloadKind {
+    Container,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum WorkloadRegistry {
+    Docker,
+    Podman,
+    Ghcr,
```

Comment:
What is Ghcr?

Reply:
Ghcr is the github registry it is very useful to keep an image linked to a repo